### PR TITLE
first draft of intro addition

### PR DESCRIPTION
**Pull request by swamidass**
I think this has my key points. Curious everyone's thoughts. 

It does need to have references added.

### CODE REVIEW

**Comment by swamidass** [comment]
@cgreene added to intro as requested.

**Comment by agitter** [comment]
Note that @cgreene is also editing the intro in #243.  We should be okay merging both pull requests because each of you is focusing on different sub-sections.

I'll read this to provide some specific feedback.

File: sections/02_intro.md [hunk 1]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
-other non-neural-network-based approaches at subsequent stages.
+[This section needs citations]
+
+Deep Learning is a collection of new techniques that together have recently demonstrated 
+breakthrough gains over existing approaches in several fields. 
+
+Deep learning is built on a very old idea, neural networks, that was first
+proposed in 1943 [doi:10.1007/BF02478259] as a model for how biological
+brains proces information. Since then, interest in neural networks a computational
+models has waxed and waned several times. This history is interesting in its own right  [@doi:10.1103/RevModPhys.34.135, @doi:10.1103/RevModPhys.34.135],
```
Comment (agitter) [inline sections/02_intro.md]: Can remove the comma between references.  Also, automatic reflow would help keep the lines to <= 80 characters, which has made it easier for us to make and review small edits to the text.

File: sections/02_intro.md [hunk 2]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+but in recent years, with the advances of Deep Learning, attention has shifted back.
+
+Several important advances make the current surge of work done in this area possible.
+
+First, several easy to use software packages (Tensorflow, Caffe, Theano) now enable a much broader range of scientists
+to build and train complicated. In the past, neural networks required very specialized knoweldge to
```
Comment (agitter) [inline sections/02_intro.md]: Missing word after "complicated"

File: sections/02_intro.md [hunk 3]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+models has waxed and waned several times. This history is interesting in its own right  [@doi:10.1103/RevModPhys.34.135, @doi:10.1103/RevModPhys.34.135],
+but in recent years, with the advances of Deep Learning, attention has shifted back.
+
+Several important advances make the current surge of work done in this area possible.
+
+First, several easy to use software packages (Tensorflow, Caffe, Theano) now enable a much broader range of scientists
+to build and train complicated. In the past, neural networks required very specialized knoweldge to 
+build and modify, including the ability to robustly code differentials of matrix 
+expressions. Errors here are often subtle and difficult to detect, so it could be 
+very difffult to tailor networks to specific problems without substantial experience
```
Comment (agitter) [inline sections/02_intro.md]: "difficult"

File: sections/02_intro.md [hunk 4]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+run very efficently on GPUs. The technology has progressed to the point that even
+algorithms can be differentiated [cite neural stack and neural memory papers].
+
+Second, key technical insight has been uncovered that guides the construction of 
+much more complicated networks that previously possible. In the past, most neural networks 
+included just a single network. A network with an arbitrary number of hidden nodes, but 
+just a single layer, can learn arbitrarily complex functions. And networks with more than one hidden
+layer (deep networks), were hard to train. However, it turns out, deep networks can more
+efficiently represent many tasks when they are built to mirror the underlying structure of the data.
+Moroever, deep networks are more robust and trainable when employing several
```
Comment (agitter) [inline sections/02_intro.md]: "Moreover"

File: sections/02_intro.md [hunk 5]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+architectural innovations: weight replication, better behaived non-linearities like rectified-linear units, residual networks,
```
Comment (agitter) [inline sections/02_intro.md]: These are all very good to point out, but I wonder whether our readers will have any idea what these mean or if they will be unintuitive jargon?

File: sections/02_intro.md [hunk 6]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+efficiently represent many tasks when they are built to mirror the underlying structure of the data.
+Moroever, deep networks are more robust and trainable when employing several
+architectural innovations: weight replication, better behaived non-linearities like rectified-linear units, residual networks, 
+and better weight initialization, and persistent memory. Likewise the central role of 
+dimensionality reduction as a strength of neural networks was elucidated, and this has
+motivated designs built to capitalize on this strength [cite autoencoders and word2vec].
+
+Third, several advances in training algorithms have enabled applications of neural networks in ways
+not obviously possible. The number of training strategies for deep learning is growing rapidly and a complete
+reveiw is beyond our scope. But these algorithms can train networks in domains where earlier algorithms struggled.
```
Comment (agitter) [inline sections/02_intro.md]: "review"

File: sections/02_intro.md [hunk 7]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+motivated designs built to capitalize on this strength [cite autoencoders and word2vec].
+
+Third, several advances in training algorithms have enabled applications of neural networks in ways
+not obviously possible. The number of training strategies for deep learning is growing rapidly and a complete
+reveiw is beyond our scope. But these algorithms can train networks in domains where earlier algorithms struggled. 
+For example, newer optimizers can very efficiently learn using batched training, where only a portion of the data
+needs to be processed at a time. These optimizers more effectively optimize very large weight vectors where many weights are only
+rarely updated. Noise constrastive error has proven particularly useful in 
+modeling language. Reinforcement learning has enabled neural networks to learn how to play games 
+like chess, GO, and poker. Curriculumn learning enables networks to gradually build up expertise to
```
Comment (agitter) [inline sections/02_intro.md]: "Curriculum"

File: sections/02_intro.md [hunk 8]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+
+First, in scientific contexts understanding the patterns in the data may be just as important as fitting the data.
+For this reason, interpretability can be more important here than other domains. Scientific work often 
+aimes to understand the underlying principles behind the data we see, and architectures and techniques that expose the
+non-obvious patterns in the data are particualrly important and very active area of research [cite examples from all sections].
+
+Second, there are important and pressing questions about how to build networks that can efficently represent 
+the underlying logic of the data. This concern of "representability" is important, because it gives insight into
+the structure of scientific data, and when understood can guide the design of more efficent and effective networks. For example,
+one particularly important study was published in Science, which demonstrates that a simple neural network can very efficiently and
```
Comment (agitter) [inline sections/02_intro.md]: I don't think we need to emphasize where it was published

File: sections/02_intro.md [hunk 9]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+accuratly model (i.e. respresent) a theoretically important quantum mechanical system [http://science.sciencemag.org/content/355/6325/602].
```
Comment (agitter) [inline sections/02_intro.md]: Glad you introduced this paper, I was looking for a place to refer to it in the review.  I think it is very relevant despite not being a biological example.

Can you please switch to the doi before merging?

File: sections/02_intro.md [hunk 10]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+
+Third, science is full of domain expertise, where there are deep traditions of thought stretching back decades and even centuries. Deep learning
+will always be in dialogue with thise expertise, to understand the key problems, encode the most salient prior knoweledge, and
```
Comment (agitter) [inline sections/02_intro.md]: "this"?

"knowledge"

File: sections/02_intro.md [hunk 11]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+understand how to judge success or failure. There is a great deal of excitement about deep learning, but in most scientific corners
+careful thought needs to be put into bringing deep learning alongside existing experts and efforts.
```
Comment (agitter) [inline sections/02_intro.md]: Is this a good place to mention the need to compare deep learning performance with existing best practices in a field?

File: sections/02_intro.md [hunk 12]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+
+Fourth, data availability and complexity is unevenly distributed accross science. Some areas of science like genomics and particle physics are swamped in petabytes and
```
Comment (agitter) [inline sections/02_intro.md]: Maybe we can keep these domain examples to biomedicine to keep in theme?  E.g. replace "chemistry" with "biochemistry" or "medicinal chemistry".

File: sections/02_intro.md [hunk 13]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+accuratly model (i.e. respresent) a theoretically important quantum mechanical system [http://science.sciencemag.org/content/355/6325/602].
+
+Third, science is full of domain expertise, where there are deep traditions of thought stretching back decades and even centuries. Deep learning
+will always be in dialogue with thise expertise, to understand the key problems, encode the most salient prior knoweledge, and
+understand how to judge success or failure. There is a great deal of excitement about deep learning, but in most scientific corners
+careful thought needs to be put into bringing deep learning alongside existing experts and efforts.
+
+Fourth, data availability and complexity is unevenly distributed accross science. Some areas of science like genomics and particle physics are swamped in petabytes and 
+exobytes of high quality data. Others, like chemistry, are comparatively data poor with well developed domain specific and effective algorithms. These
+differences become consequential and define the most successul approachs. For example, the convergence of lower amounts of data
```
Comment (agitter) [inline sections/02_intro.md]: "approaches"

File: sections/02_intro.md [hunk 14]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+and important nuances to the domain might favor lower parameter networks that incorporate domain specific knowledge and fuse data of multiple
+different types. This flexibility, remember, is one of the most striking strengths of neural networks. In the long run,
+it is an open question the most effect strategies will be, but in this time of creative experimenation optimism is justified.
+
+None of these scientific concerns should dampen enthusiasm about deep learning. Rather, because the approaches flexibility,
+there is good reason to believe that carefully defined networks might enbable important scientific advances.
```
Comment (agitter) [inline sections/02_intro.md]: "enable"

File: sections/02_intro.md [hunk 15]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+careful thought needs to be put into bringing deep learning alongside existing experts and efforts.
+
+Fourth, data availability and complexity is unevenly distributed accross science. Some areas of science like genomics and particle physics are swamped in petabytes and 
+exobytes of high quality data. Others, like chemistry, are comparatively data poor with well developed domain specific and effective algorithms. These
+differences become consequential and define the most successul approachs. For example, the convergence of lower amounts of data
+and important nuances to the domain might favor lower parameter networks that incorporate domain specific knowledge and fuse data of multiple
+different types. This flexibility, remember, is one of the most striking strengths of neural networks. In the long run,
+it is an open question the most effect strategies will be, but in this time of creative experimenation optimism is justified.
+
+None of these scientific concerns should dampen enthusiasm about deep learning. Rather, because the approaches flexibility,
```
Comment (agitter) [inline sections/02_intro.md]: Something missing in "approaches flexibility"

File: sections/02_intro.md (hunk 14 above)
Comment (swamidass) [inline sections/02_intro.md]: Sorry about the delay on this. I just submitted a grant yesterday and am returning here. Will fix soon.

File: sections/02_intro.md [hunk 16]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+
+Deep learning is built on a very old idea, neural networks, that was first
+proposed in 1943 [doi:10.1007/BF02478259] as a model for how biological
+brains proces information. Since then, interest in neural networks a computational
+models has waxed and waned several times. This history is interesting in its own right  [@doi:10.1103/RevModPhys.34.135, @doi:10.1103/RevModPhys.34.135],
+but in recent years, with the advances of Deep Learning, attention has shifted back.
+
+Several important advances make the current surge of work done in this area possible.
+
+First, several easy to use software packages (Tensorflow, Caffe, Theano) now enable a much broader range of scientists
```
Comment (cgreene) [inline sections/02_intro.md]: This is an important point that we shouldn't lose.

File: sections/02_intro.md [hunk 17]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+and better weight initialization, and persistent memory. Likewise the central role of 
+dimensionality reduction as a strength of neural networks was elucidated, and this has
+motivated designs built to capitalize on this strength [cite autoencoders and word2vec].
+
+Third, several advances in training algorithms have enabled applications of neural networks in ways
+not obviously possible. The number of training strategies for deep learning is growing rapidly and a complete
+reveiw is beyond our scope. But these algorithms can train networks in domains where earlier algorithms struggled. 
+For example, newer optimizers can very efficiently learn using batched training, where only a portion of the data
+needs to be processed at a time. These optimizers more effectively optimize very large weight vectors where many weights are only
+rarely updated. Noise constrastive error has proven particularly useful in
```
Comment (cgreene) [inline sections/02_intro.md]: contrastive

File: sections/02_intro.md [hunk 18]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+not obviously possible. The number of training strategies for deep learning is growing rapidly and a complete
+reveiw is beyond our scope. But these algorithms can train networks in domains where earlier algorithms struggled. 
+For example, newer optimizers can very efficiently learn using batched training, where only a portion of the data
+needs to be processed at a time. These optimizers more effectively optimize very large weight vectors where many weights are only
+rarely updated. Noise constrastive error has proven particularly useful in 
+modeling language. Reinforcement learning has enabled neural networks to learn how to play games 
+like chess, GO, and poker. Curriculumn learning enables networks to gradually build up expertise to
+solve particularly challenging algorithmic problems. Dropout nodes and layers make networks much more
+robust, even when the number of weights are dramatically increased.
+
```
Comment (cgreene) [inline sections/02_intro.md]: There is quite a bit of jargon in here. If it's going to be included, citations need to exist to point readers towards a resource for each topic.

File: sections/02_intro.md [hunk 19]
```diff
@@ -34,41 +34,88 @@ learning methods more challenging or less fruitful to apply.
...
+Given the intrinsic flexibility of deep learning, it is important to consider the specific values and goals
+that are particularly important in scientific inquiry.
+
+First, in scientific contexts understanding the patterns in the data may be just as important as fitting the data.
+For this reason, interpretability can be more important here than other domains. Scientific work often 
+aimes to understand the underlying principles behind the data we see, and architectures and techniques that expose the
+non-obvious patterns in the data are particualrly important and very active area of research [cite examples from all sections].
+
+Second, there are important and pressing questions about how to build networks that can efficently represent 
+the underlying logic of the data. This concern of "representability" is important, because it gives insight into
```
Comment (cgreene) [inline sections/02_intro.md]: logic -> structure?

File: sections/02_intro.md (hunk 12 above)
Comment (cgreene) [inline sections/02_intro.md]: It's not just the _amount_ of data but the complexity of the required features + number of examples. I worry about focusing solely on scale.

File: sections/02_intro.md (hunk 3 above)
Comment (agapow) [inline sections/02_intro.md]: Spelling: "knowledge"

File: sections/02_intro.md (hunk 7 above)
Comment (agapow) [inline sections/02_intro.md]: "These optimizers more effectively optimize" feels very awkward. Perhaps "These approaches more effectively optimize" or "These optimizers are more effective where"

File: sections/02_intro.md (hunk 10 above)
Comment (agapow) [inline sections/02_intro.md]: "accurately"

File: sections/02_intro.md (hunk 18 above)
Comment (agapow) [inline sections/02_intro.md]: "Curriculum"

File: sections/02_intro.md (hunk 16 above)
Comment (enricoferrero) [inline sections/02_intro.md]: Agree. Should we cite these with URLs though? Also maybe 'deep learning frameworks' better than 'software packages'. I'd add MxNet (and maybe even h20.ai) to that list.

**Comment by enricoferrero** [comment]
The May 1st deadline is fast approaching. My suggestion would be either to merge this as is to allow further edits or follow @cgreene advice of rejecting and including some of the themes presented in a new PR.

**Comment by cgreene** [comment]
In the interest of making progress on this I have simply moved the merge forward by including the additional text alongside the previous text. The intro needs to be revamped, and this will let us move things forward. We're going to have to do some pretty extensive editing here to make things work. But at least we can move ahead.

**Comment by agitter** [comment]
@enricoferrero I agree.  Because @swamidass has not come back to revise this, we will close this pull request.  @cgreene or I will select parts of the text here to merge into the current draft of the intro, making smaller, more local edits.

**Comment by agitter** [comment]
_(deleted my last comment because @cgreene took care of the merge)_

To confirm I'm reading the merged version correctly, the draft lead by `One version` came from this pull request and the draft lead by `Another version` is the existing text?

**Comment by cgreene** [comment]
@agitter - yep that's it!

**Pull request merged**